src/Json.php

```diff
@@ -46,8 +46,15 @@ public static function decode($data)
 
         $result = json_decode($data, true);
 
-        if (json_last_error() !== JSON_ERROR_NONE) {
-            throw new LogicException(self::getLastJsonError());
+        if (JSON_ERROR_NONE !== json_last_error()) {
+            $lastError = self::getLastJsonError();
+            $payload = substr($data, 0, 255);
+
+            if (strlen($data) > 255) {
+                $payload .= '... (truncated)';
+            }
+
+            throw new LogicException(sprintf('%s on \'%s\'', $lastError, $payload));
```

Comment:
I would probably change this to `occurred on` or `occurred in`.

Reply:
Most "Syntax Error" messages on compilers use "on", followed by the affected line and contents. Why change that?

Reply:
Actually in this case simply changing `on` to `in` would make it follow PHP. The `occurred` part was something that popped in my head, but isn't necessary.

For example:

```
>>> json_code()
PHP Fatal error:  Call to undefined function json_code() in eval()'d code on line 1
```